File: tests/phpunit/includes/LuaLibrary/lua/pure/picklelib/report/ConstituentTest.lua
--- Tests for the constituent module
-- This is a preliminary solution
-- @license GNU GPL v2+
-- @author John Erling Blad < jeblad@gmail.com >


local testframework = require 'Module:TestFramework'

local lib = require 'picklelib/report/Constituent'
local name = 'constituent'
local class = 'constituent'

local function makeTest( ... )
	return lib.create( ... )
end

local function testExists()
	return type( lib )
end

local function testCreate( ... )
	return type( makeTest( ... ) )
end

local function testRealize( ... )
	return makeTest( ... ):realize()
end

local function testType( ... )
	return makeTest( ... ):type()
end

local tests = {
	{ name = name .. ' exists', func = testExists, type='ToString',
	  expect = { 'table' }
	},
	{ name = name .. '.create (nil value type)', func = testCreate, type='ToString',
	  args = { nil },
	  expect = { 'table' }
	},
	{ name = name .. '.create (single value type)', func = testCreate, type='ToString',
	  args = { 'a' },
	  expect = { 'table' }
	},
	{ name = name .. '.create (multiple value type)', func = testCreate, type='ToString',
	  args = { 'a', 'b', 'c' },
	  expect = { 'table' }
	},
	{ name = name .. '.realize (nil)', func = testRealize,
	  args = { nil },
	  expect = { '' }
	},
	{ name = name .. '.type ()', func = testType,
	  expect = { class }
	},
}

return testframework.getTestProvider( tests )

File: tests/phpunit/includes/LuaLibrary/lua/pure/picklelib/report/ConstituentTest.php
<?php

namespace Pickle\Test;

use Scribunto_LuaEngineTestBase;

/**
 * @group Pickle
 *
 * @license GNU GPL v2+
 *
 * @author John Erling Blad < jeblad@gmail.com >
 */
class ConstituentTest extends Scribunto_LuaEngineTestBase {

	protected static $moduleName = 'ConstituentTest';

	function getTestModules() {
		return parent::getTestModules() + [
			'ConstituentTest' => __DIR__ . '/ConstituentTest.lua'
		];
	}
}
